internal/pokt_client_decorators/cached_client_test.go

```diff
@@ -0,0 +1,204 @@
+package pokt_client_decorators
+
+// Basic imports
+import (
+	"errors"
+	"os-gateway/mocks"
+	"os-gateway/pkg/pokt/pokt_v0/models"
+	"testing"
+	"time"
+
+	"github.com/jellydator/ttlcache/v3"
+	"github.com/stretchr/testify/suite"
+)
+
+type CachedClientTestSuite struct {
+	suite.Suite
+	mockPocketService    *mocks.PocketService
+	newCachedClient      *CachedClient
+	mockTTLCachedService *mocks.TTLCacheService[string, *models.GetSessionResponse]
+}
+
+func (suite *CachedClientTestSuite) SetupTest() {
+	suite.mockPocketService = new(mocks.PocketService)
+	suite.mockTTLCachedService = new(mocks.TTLCacheService[string, *models.GetSessionResponse])
+	suite.newCachedClient = NewCachedClient(suite.mockPocketService, suite.mockTTLCachedService)
+}
+
+// test GetSession using table driven tests
+func (suite *CachedClientTestSuite) TestGetSession() {
+
+	testRequest := &models.GetSessionRequest{
+		AppPubKey: "test",
+		Chain:     "test",
+		Height:    1,
+	}
+
+	ttlcacheItem := &ttlcache.Item[string, *models.GetSessionResponse]{}
+
+	testResponse := &models.GetSessionResponse{}
+
+	errUnderlayingProvider := errors.New("error underlaying provider")
+
+	// create test cases
+	testCases := []struct {
+		name             string
+		setupMocks       func()
+		expectedResponse *models.GetSessionResponse
+		expectedError    error
+	}{
+		{
+			name: "NotCached",
```

Comment:
This scenario is testing if the grabbing the session recently failed, not if it's not "cached". Please rename to reflect